src/Command/Dedupe.php

```diff
@@ -74,6 +65,49 @@ protected function execute(InputInterface $input, OutputInterface $output): void
         }
     }
 
+    /**
+     * @param string[] $fields
+     */
+    private function getValuesHash(array $fields, array $data): string
+    {
+        $values = [];
+        foreach ($fields as $field) {
+            $values[$field] = $data[$field];
+        }
+
+        return sha1(serialize($values));
+    }
+
+    /**
+     * @param string[] $fields
+     * @param bool[]   $seenValues Hashmap of value-hash => bool, to keep track of values already encountered
+     */
+    private function handleDuplicate(array $fields, array &$message, array &$seenValues, int $messageNum): ?array
+    {
+        $payload = json_decode($message['payload'], true);
+
+        // Check that all given fields exists in the payload:
+        foreach ($fields as $field) {
+            if (!array_key_exists($field, $payload)) {
+                throw new \UnexpectedValueException(sprintf(
+                    'Payload of message #%d does not have the required fields %s (%s)',
```

Comment:
i know this is getting complicated, but should we say fields can also be optional?

Reply:
I'd leave this for a next MR, we could define fields as `?field` to make them optional :)